### ISSUE TITLE
Necesity of ANSYS product license to execute PyDYNA

### ISSUE THREAD

**Issue by Taniguchi-Shin** [ansys/pydyna#536]
Dear All,

This is Shin Taniguchi of JSOL Corporation.
I'm trying to build PyDYNA based on the virtual environment with Ansys Python Manager.
And I have already installed Ansys product whose version is 2024R1.

Is the ANSYS product license needed to execute PyDYNA?
I have already anabled LSTC_LICENSE_SERVER to execute Ansys LS-DYNA modules.

Now, I have suceeded to execute PyDYNA without the connection of the ANSYS product license.
I feel that these server systems (pre/solver) don't go through ANSYS product's objects.

Best Regards,
Shin Taniguchi/JSOL

JSOL Corporation, CAE Technology Department,
Engineering Technology Division.
KUDAN-KAIKAN TERRACE 11th Floor
1-6-5 Kudanminami, Chiyoda-ku, Tokyo, 102-0074 Japan
E-Mail: [taniguchi.shin@jsol.co.jp](mailto:taniguchi.shin@jsol.co.jp)
Phone: +81-80-2353-9121(Direct), +81-3-6261-7168

**Comment by Taniguchi-Shin** [comment]
Sorry, I'm in a hurry, so please reply.

**Comment by RobPasMue** [comment]
@ansys/pydyna please respond

**Comment by zhangzhanqun** [comment]
Hi Taniguchi,
You are right, the Ansys product license is needed to execute PyDYNA.
Actually,the PyDYNA-pre is not need the ANSYS license,
 but the PyDYNA-server will use the executable file in your installed Ansys 2024R1.So the PyDYNA-server need the ANSYS product license.

**Comment by Taniguchi-Shin** [comment]
Dear Zhangzhanqun,

This is Shin Taniguchi of JSOL.
Thank you for your reply.

I understand that PyDYNA has PyDYNA-pre-sever and PyDYNA-solver-server.
I got PyDYNA-solver-server from https://github.com/ansys/pydyna site.

I have attached python script in PyDYNA-solver-server system.
Based on this script, can we change it without installed Ansys 2024R1?
I have only found the module path description on the script.

Best Regards,
Shin Taniguchi/JSOL

[server.zip](https://github.com/user-attachments/files/17456552/server.zip)

**Comment by zhangzhanqun** [comment]
Hi Taniguchi, It is necessary to install the Ansys package. you can find that there are function named "subprocess.Popen" in server python script, it is used to call the executable file in Ansys package.

**Comment by Taniguchi-Shin** [comment]
Dear Zhangzhanqun, Thank you for your reply.

I have found "subprocess.Popen" in server python script.
I guess that this is used to call LS-DYNA modules in the Ansys procut's instalation folder.

So, when these discriptions are modified correctly, we can execute PyDYNA without Ansys product license.
I will investigate these presumptions internally.
If you have any comments about them, please let me know.

Otherwise, we're concerned about the possibility of conflict with Ansys product's commercial process.

Best Regards,
Shin Taniguchi/JSOL

**Comment by Taniguchi-Shin** [comment]
Sorry, could you comment for above response?

**Comment by RobPasMue** [comment]
@Taniguchi-Shin - I believe @zhangzhanqun already answered to all your questions. What else do you need his input on?

**Comment by Taniguchi-Shin** [comment]
Dear RobPasMue and Zhangzhanqun,

Sorry, my explanation isn't enough.
I guess that "subprocess.Popen" in the PyDYNA solver-server python script is only used to call LS-DYNA modules such as lsdyna_dp.exe.
So, discriptions determining the folder path of LS-DYNA modules are modified correctly, Ansys product license isn't needed also PyDYNA pre-server.

Is there another purpose of "subprocess.Popen" to call LS-DYNA modules?
If it's "yes", the necessity of Ansys product license to execute PyDNYA pre/solver-server is plausible.

Best Regards,
Shin Taniguchi/JSOL

**Comment by zhangzhanqun** [comment]
Hi Taniguchi,

    You are right, the "subprocess.Popen" is used to call LS-DYNA modules, Ansys product license is needed for LS-DYNA modules, in other words, the lsdyna_dp.exe can not be run if the Ansys product license is not setup properly.

Regards,
zhanqun.

**Comment by Taniguchi-Shin** [comment]
Dear RobPasMue and Zhangzhanqun,

Thank you for your reply.

Now, we're connecting ANSYS License server with internal floating license.
Please see an attached image file.
But, although the licensing setting isn't enabled, PyDYNA solver-server can be executed.

> the lsdyna_dp.exe can not be run if the Ansys product license is not setup properly.
This explanation is conflicted with our current process.
Could you confirm this matter? 

![Image](https://github.com/user-attachments/assets/678fbbbd-9a5c-4f48-87a2-9d556385f577)

Best Regards,
Shin Taniguchi/JSOL

**Comment by RobPasMue** [comment]
@Taniguchi-Shin - please delete the above image since it might contain confidential information

**Comment by Taniguchi-Shin** [comment]
Sorry. I have just deleted the confidential information.

**Comment by Taniguchi-Shin** [comment]
Dear RobPasMue and Zhangzhanqun,
Could you confirm this issue about Ansys product license required?

Now, we can execute pydyna pre/solver module without ANSYS License server (FlexNet) connection.
Otherwise, we have already installed Ansys product 2024R1 and pydyna scripts clearly call it's insalled folder.

We guess following constriction. Is this true?
- We need to have at least one version of Ansys product and purchase at least one license to use pydyna system.

If this is "yes", we will introduce it to JSOL users if they question about pydyna to us.

Best Regards,
Shin Taniguchi/JSOL

**Comment by Taniguchi-Shin** [comment]
Dear RobPasMue and Zhangzhanqun, Sorry for late response.

Now, one of our users questions about PyDYNA and PyDPF.
We have to clear the neccesity of Ansys product license to execute PyDYNA for user support.

Now, we understand that Ansys product installation is needed to execute PyDYNA pre/solver server.
Because python script refers the installation folder of Ansys product.
And this means that Ansys product license is needed to consider the post processing such as PyDPF.

Is this recognition ture?
And coud you comment as soon as possible?

Best Regards,
Shin Taniguchi/JSOL

**Comment by Taniguchi-Shin** [comment]
Now, we have already understood that PyDPF needs Ansys product license at least.

**Comment by Taniguchi-Shin** [comment]
Sorry, could you comment for above response?
We're in a bit of a hurry for our sales.

**Comment by zhangzhanqun** [comment]
@Taniguchi-Shin Yes, you are right, the Ansys product installation is needed to execute PyDYNA pre/solver server.

**Comment by Taniguchi-Shin** [comment]
Dear Zhangzhanqun,

This is Shin Taniguchi of JSOL. Thank you for your reply.
Now, in our environment, we can execute PyDYNA pre/solver server without the connection of ANSYS License server.
Do you have the plan to change above specification?

Best Regards,
Shin Taniguchi/JSOL

**Comment by zhangzhanqun** [comment]
Hi @Taniguchi-Shin PyDYNA pre/solver server can be executed without the connection of ANSYS License server, But in PyDYNA solver,  if we call  LS-DYNA to run keyword file, then we need the ANSYS License server.

**Comment by Taniguchi-Shin** [comment]
Dear Zhangzhanqun,
Thank you for your reply.

We think that we need Ansys LS-DYNA license server to call LS-DYNA to run keyword file.
And, we think that Ansys LS-DYNA license server is indipendent of ANSYS product's license server executed with following system on Windows.

<img width="362" alt="Image" src="https://github.com/user-attachments/assets/a95fde36-f6f1-4ca7-8867-79b14a98f706" />

Is this true?
Please let me a reason of neccesity of ANSYS license server if this is false.

**Comment by zhangzhanqun** [comment]
Hi @Taniguchi-Shin Actually, if you have LS-DYNA license to make sure LS-DYNA can be run, then the ANSYS license is not necessary.

**Comment by Taniguchi-Shin** [comment]
Dear Zhangzhanqun,
Thank you for your reply. I could understand them.

Based on using PyDPF, ANSYS product's license is needed on either way.
So, we understand that PyDYNA execution through post-processing needs ANSYS product's license actually.
Thank you for your explanation.

Best Regards,
Shin Taniguchi/JSOL

**Comment by Taniguchi-Shin** [comment]
Dear Zhangzhanqun, please close this issue.
Thsnk you for your explanation.
